advanced_message_passing.md

```diff
@@ -1,8 +1,120 @@
 ## Advanced Message Passing
 
-* Ephemeral - URL / QR 
+TODO:
+
 * Deep linking
 * NFC
 * Wifi-Direct
 * Bluetooth
 * Sneakernet / file oriented / mimetypes
+
+#### URL & QR Codes
+
+When passing a DIDComm Message between two parties, it is often useful to present a message in the form of a URL or encoded into the form of a QR code for scanning with a smartphone or other camera. The format for a QR code is simply the encoded URL form of a message.
+
+##### Privacy Considerations
+
+Any information passed via a URL or QR code is unencrypted, and may be observed by another party. This lack of privacy must be minded in two different ways.
+
+First, no private information may be passed in the message. Private information should be passed between parties in encrypted messages only. Any protocol message that contains private information should not be passed via URL or QR code.
+
+Second, any identifiers passed in a message sent via URL or QR code must no longer be considered private. Any DID used or other identifier no longer considered private MUST be rotated over a secure connection if privacy is required.
+
```

Comment:
```suggestion
Additionally, because of the way URL shortening services work, using these services are SHOULD NOT be used because it allows the services to track and identify the recipient who's receiving the message based on IP, time of day, and frequency of message retrieval.
```

Reply:
This is slightly redundant with the warning in about message retrieval (name different than URL shortening on purpose) at the bottom of the doc.  Should we at least include a reference to that section in this warning? Thoughts?